src/component/CalculatorDigit.js

```diff
@@ -0,0 +1,36 @@
+import CalculatorDigitItem from "./CalculatorDigitItem";
+const DIGIT = ["9", "8", "7", "6", "5", "4", "3", "2", "1", "0"];
+
+const CalculatorDigit = ({ totalState, countState }) => {
+  const onClickNumber = (num) => {
+    countNumber();
+    if (totalState.total === "0") {
+      totalState.setTotal(num);
+    } else {
+      totalState.setTotal(totalState.total + num);
+    }
```

Comment:
```suggestion
    const newTotal = totalState.total === "0" ? num :  totalState.total + num;
    totalState.setTotal(newTotal);
```

가능하다면 제어"문"보다는 표현"식"을 위주로 코드를 작성하시는걸 추천드립니다.
가독성도 좋고, 항상 새로운 값을 생성하기 때문에 흐름을 좇아가기가 더 편합니다 😄